llvm/unittests/AsmParser/AsmParserTest.cpp

```diff
@@ -479,4 +483,60 @@ TEST(AsmParserTest, DIExpressionBodyAtBeginningWithSlotMappingParsing) {
   ASSERT_EQ(Mapping.MetadataNodes.size(), 0u);
 }
 
+#define ASSERT_EQ_LOC(Loc1, Loc2)                                              \
+  do {                                                                         \
+    bool AreLocsEqual = Loc1.contains(Loc2) && Loc2.contains(Loc1);            \
+    if (!AreLocsEqual) {                                                       \
+      dbgs() << #Loc1 " location: " << Loc1.Start.Line << ":"                  \
+             << Loc1.Start.Col << " - " << Loc1.End.Line << ":"                \
+             << Loc1.End.Col << "\n";                                          \
+      dbgs() << #Loc2 " location: " << Loc2.Start.Line << ":"                  \
+             << Loc2.Start.Col << " - " << Loc2.End.Line << ":"                \
+             << Loc2.End.Col << "\n";                                          \
```

Comment:
please wrap these line with LLVM_DEBUG. Though it's a bit unusual to have debug prints in a unit test

Reply:
gtest has it's own support for printing additional debugging information if an expectation fails. In the simplest case you can write something like `ASSERT_TRUE(AreLocsEqual) << ...`. As this test is not hot, you shouldn't need the more sophisticated methods.

Reply:
Also EXPECT seems more appropriate than ASSERT here.